cpp/src/arrow/flight/sql/odbc/flight_sql/flight_sql_statement_get_tables.cc

```diff
@@ -31,6 +32,15 @@ using arrow::flight::FlightClientOptions;
 using arrow::flight::FlightInfo;
 using arrow::flight::sql::FlightSqlClient;
 
+static void AddTableType(std::string& table_type, std::vector<std::string>& table_types) {
```

Comment:
Good catch. I remember this bug. When I was testing before, I put down `"BASE TABLE"` and it didn't work, so I used single quotes to wrap `"'BASE TABLE'"` which worked, so I thought it was fine. But by checking [SQLTables doc](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqltables-function?view=sql-server-ver17#comments), each keyword can be quoted or unquoted, so this is a bug for sure.
It's good that we are fixing this now